lib/register-loader.js

```diff
@@ -0,0 +1,27 @@
+const { register } = require('node:module');
+const { join } = require('node:path');
+const { pathToFileURL } = require('node:url');
+const semver = require('semver');
+
+const { send } = require('./ipc');
+const localPath = require('./local-path');
+const resolveMain = require('./resolve-main');
+
+exports.registerLoader = () => {
+  if (!semver.satisfies(process.version, '>=20.6.0')) return;
+
+  const loaderURL = pathToFileURL(resolveMain(localPath(join('loaders', 'load.mjs'))));
+
+  const { port1, port2 } = new MessageChannel();
+  port1.on('message', ({ required } = {}) => {
+    send({ required });
+  });
```

Comment:
Instead of adding ˙process.exit(0)` calls, the better solution would be to just call `port1.unref()` (https://nodejs.org/api/worker_threads.html#portunref).